src/erc7432/role/grant-handler.ts

```diff
@@ -0,0 +1,50 @@
+import { log } from '@graphprotocol/graph-ts'
+import { RoleGranted } from '../../../generated/Roles/ERC7432Roles'
+import { Account, Nft, Role } from '../../../generated/schema'
+import { generateId } from '../../utils/helper'
+
+export function handleRoleGranted(event: RoleGranted): void {
+  const tokenId = event.params._tokenId.toString()
+  const tokenAddress = event.params._tokenAddress.toHexString()
+
+  const nftId = generateId(tokenId, tokenAddress)
+  const nft = Nft.load(nftId)
+
+  if (!nft) {
+    log.warning('[handleRoleGranted] NFT {} does not exist, skipping transfer...', [nftId])
+    return
+  }
+
+  const address = event.address.toHexString()
+  const nftOwner = nft.owner
+  const grantor = Account.load(address)
+
+  if (!grantor || nftOwner != address) {
+    log.warning('[handleRoleGranted] NFT {} is not owned by {}, skipping transfer...', [nftId, address])
+    return
+  }
```

Comment:
I would separate this in two different logs (it's helpfull for debugging):

```suggestion
  if (!grantor) {
    log.warning('[handleRoleGranted] grantor {} does not exist, skipping transfer...', [address])
    return
  }
  
  if (grantor.id != nft.owner) {
    log.warning('[handleRoleGranted] NFT {} is not owned by {}, skipping transfer...', [nftId, grantor.id.toString()])
    return
  }
```